sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -639,8 +639,8 @@ abstract class TypeCoercionBase {
         s.copy(right = Cast(s.right, s.left.dataType))
       case s @ SubtractTimestamps(AnyTimestampType(), AnyTimestampType(), _, _)
         if s.left.dataType != s.right.dataType =>
-        val newLeft = castIfNotSameType(s.left, AnyTimestampType.defaultConcreteType)
-        val newRight = castIfNotSameType(s.right, AnyTimestampType.defaultConcreteType)
+        val newLeft = castIfNotSameType(s.left, TimestampWithoutTZType)
```

Comment:
Here the result is the same if casting both to `TimestampWithoutTZType` or `TimestampType`.  Casting to TimestampWithoutTZType is more straightforward.